crates/rpc/lean/src/lib.rs

```diff
@@ -1,14 +1,55 @@
-pub fn add(left: u64, right: u64) -> u64 {
-    left + right
+use std::sync::Arc;
+
+use actix_web::{App, HttpServer, dev::ServerHandle, middleware, web::Data};
+use config::LeanRpcServerConfig;
+use ream_chain_lean::lean_chain::LeanChain;
+use tokio::sync::RwLock;
+use tracing::info;
+
+use crate::routes::register_routers;
+
+pub mod config;
+pub mod handlers;
+pub mod routes;
+
+/// Start the Lean API server.
+pub async fn start_lean_server(
+    server_config: LeanRpcServerConfig,
+    lean_chain: Arc<RwLock<LeanChain>>,
+) -> std::io::Result<()> {
+    info!(
+        "starting HTTP server on {:?}",
+        server_config.http_socket_address
+    );
+    let stop_handle = Data::new(StopHandle::default());
+
+    let server = HttpServer::new(move || {
+        let stop_handle = stop_handle.clone();
+        App::new()
+            .wrap(middleware::Logger::default())
+            .app_data(stop_handle)
+            .app_data(Data::new(lean_chain.clone()))
+            .configure(register_routers)
+    })
+    .bind(server_config.http_socket_address)?
+    .run();
+
+    server.await
+}
+
+#[derive(Default)]
+struct StopHandle {
+    inner: parking_lot::Mutex<Option<ServerHandle>>,
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+#[allow(dead_code)]
+impl StopHandle {
+    pub(crate) fn register(&self, handle: ServerHandle) {
+        *self.inner.lock() = Some(handle);
+    }
 
-    #[test]
-    fn it_works() {
-        let result = add(2, 2);
-        assert_eq!(result, 4);
+    pub(crate) fn stop(&self, graceful: bool) {
+        #[allow(clippy::let_underscore_future)]
+        let _ = self.inner.lock().as_ref().unwrap().stop(graceful);
```

Comment:
something really seems wrong here if we need ``#[allow(clippy::let_underscore_future)]``, I think that clippy allowed should be removed, but if you decide to delete StopHandle altogether that is good too